Convert internals to use Moo and Moo compatible types

kentfredric opened the pull request:
_mostly_ straightforward porting. 

Not sure about the type logic exactly though.

N.B. Third Party `use` declarations performed intentionally **BEFORE** `use Moo::Role`, as this is the standard work-around to prevent `Moo::Role` treating `import()`'d subs as methods that need to be reapplied to child classes.

kentfredric committed:
Convert internals to use Moo and Moo compatible types

rjbs commented:
This may wait a while, as this kind of change will result in major amounts of re-testing needed on $Work code.

rjbs commented:
willing to apply, but it no longer applies

kentfredric commented:
I've rebased it and fixed up what was breaking and it should still be consistent.

Though I'm sorely tempted to rip out Type::Tiny and just hand-write all the ->isa()'s

kentfredric commented:
Ugh, there's still some Class::MOP code in here that was one of those hidden dependencies. 

I'm not sure what the alternative is here.

https://github.com/rjbs/Role-Identifiable/blob/master/lib/Role/Identifiable/HasTags.pm#L61

kentfredric committed:
Remove dependency on Class::MOP by using MRO directly to find methods

kentfredric committed:
Cease using Moose in tests.

kentfredric commented:
I've hacked some changes in a separate branch that does:
- Removes Types::Standard usage by hand-coding the logic ( which could be better ? _shrug_ )
- Hand-codes the Class::MOP logic using mro methods 

It seems a little more experimentally, but I can squash it into this branch as needed.

https://github.com/kentfredric/role-identifiable/compare/mooify-notypes

kentfredric commented:
Now only the "No Types::Tiny" stuff is in the other branch. The rest of it is now verifiably running tests without loading moose =)
